src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1317,13 +1317,6 @@ public void tableChanged(TableModelEvent e) {
             detailsTable.addFocusListener(repaintListener);
         }
 
-        // TAB/SHIFT-TAB should transfer focus and ENTER should select an item.
-        // We don't want them to navigate within the table
```

Comment:
what was the original purpose of this code? it seems like the code was placed here for a reason but maybe it's no longer relevant?

Reply:
Yes, I hope now it is safe to remove these lines. Since TAB/SHIFT navigation is rejected with this [fix/line](https://github.com/openjdk/jdk/blame/5528ad74902fa4f4ec621d70e7e7d85f4ac1d780/src/java.desktop/share/classes/sun/swing/FilePane.java#L1314) which is further handled in BasicTableUI class ([Here](https://github.com/openjdk/jdk/blame/7b38bfea331437ad99277032de7fce939303abc8/src/java.desktop/share/classes/javax/swing/plaf/basic/BasicTableUI.java#L679)).

Reply:
It's not about `JFileChooser` TAB/ENTER key functionality here. The issue is w.r.t to JTable which when used/invoked after `JFileChooser` `DetailsView` is clicked. The functionality of TAB/ENTER which is supposed to move the focus to next column/row is been removed in `DetailsView` of `JFileChooser`. As in test case, I have added `JFileChooser` and `JTable` in a single frame. Once user clicks on `DetailsView` of `JFileChooser` and then tries to navigate within `JTable`, the functionality doesn't work since it's been removed from `UIActionMap`.